In [1]:
import mysql.connector

# Connect to the database
connection = mysql.connector.connect(
    host="localhost",
    user="root",
    password="password",
    database="yianshui"
)

cursor = connection.cursor()

# Query to get all table names
cursor.execute("SHOW TABLES")
tables = cursor.fetchall()

# Display the table names
print("Tables in the database:")
for table in tables:
    print(table[0])

cursor.close()


Tables in the database:
alembic_version
company_info
company_reports
users


True

In [14]:
table_name = input("Enter the table name to view data: ")

cursor = connection.cursor(dictionary=True)

# Fetch data from the selected table
cursor.execute(f"SELECT * FROM {table_name} LIMIT 10")  # Limit to 10 rows for viewing
rows = cursor.fetchall()

# Display the data
print(f"Data from {table_name}:")
for row in rows:
    print(row)

cursor.close()


Data from company_info:
{'id': 21, 'company_name': '', 'tax_number': 'TaxpayerId002', 'post_data': '{"dateSource": 0, "dateType": 0, "year": 2022, "taxpayerNo": "TaxpayerId002", "timestamp": "2024-12-05T18:43:53.867597"}', 'post_initiator_user_id': 5, 'status': 1, 'query_result': None, 'index_standard_type': '', 'industry': '', 'registration_type': '', 'taxpayer_nature': '', 'upload_year': 2022, 'uploaded_files': '["企业所得税年度纳税申报表_2021.xlsx", "企业所得税年度纳税申报表_2022.xlsx", "增值税纳税申报表_2021.xls", "增值税纳税申报表_2022.xls", "财务报表_2021.xlsx", "财务报表_2022.xlsx"]', 'created_at': datetime.datetime(2024, 12, 5, 18, 43, 53)}
{'id': 22, 'company_name': '易安税联调测试企业', 'tax_number': 'TaxpayerId006', 'post_data': '{"dateSource": 0, "dateType": 0, "year": 2022, "taxpayerNo": "TaxpayerId006", "timestamp": "2024-12-05T18:58:34.314037"}', 'post_initiator_user_id': 4, 'status': 1, 'query_result': None, 'index_standard_type': '1', 'industry': '0', 'registration_type': '0', 'taxpayer_nature': '0', 'upload_year': 2022, 'up

True

#### company_report

{'id': 4, 'user_id': 5, 'company_tax_number': 'TaxpayerId002', 'report_type': 'annual', 'year': 2022, 'month': None, 'quarter': None, 'report_data': '{"riskMain": {"id": 4965, "tenant": "1008", "dateTime": 2022, "noRiskNum": 36, "lowRiskNum": 2, "midRiskNum": 6, "totalScore": "72.40", "highRiskNum": 9, "queryCondition": 20220100, "abnormalRiskNum": 7}}', 'created_at': datetime.datetime(2024, 12, 5, 18, 44, 3), 'updated_at': None}
{'id': 5, 'user_id': 4, 'company_tax_number': 'TaxpayerId006', 'report_type': 'annual', 'year': 2022, 'month': None, 'quarter': None, 'report_data': '{"riskMain": {"id": 4966, "tenant": "1012", "dateTime": 2022, "noRiskNum": 36, "lowRiskNum": 2, "midRiskNum": 6, "totalScore": "72.40", "highRiskNum": 9, "queryCondition": 20220100, "abnormalRiskNum": 7}}', 'created_at': datetime.datetime(2024, 12, 5, 18, 59, 1), 'updated_at': None}

#### company_info

{'id': 21, 'company_name': '', 'tax_number': 'TaxpayerId002', 'post_data': '{"dateSource": 0, "dateType": 0, "year": 2022, "taxpayerNo": "TaxpayerId002", "timestamp": "2024-12-05T18:43:53.867597"}', 'post_initiator_user_id': 5, 'status': 1, 'query_result': None, 'index_standard_type': '', 'industry': '', 'registration_type': '', 'taxpayer_nature': '', 'upload_year': 2022, 'uploaded_files': '["企业所得税年度纳税申报表_2021.xlsx", "企业所得税年度纳税申报表_2022.xlsx", "增值税纳税申报表_2021.xls", "增值税纳税申报表_2022.xls", "财务报表_2021.xlsx", "财务报表_2022.xlsx"]', 'created_at': datetime.datetime(2024, 12, 5, 18, 43, 53)}
{'id': 22, 'company_name': '易安税联调测试企业', 'tax_number': 'TaxpayerId006', 'post_data': '{"dateSource": 0, "dateType": 0, "year": 2022, "taxpayerNo": "TaxpayerId006", "timestamp": "2024-12-05T18:58:34.314037"}', 'post_initiator_user_id': 4, 'status': 1, 'query_result': None, 'index_standard_type': '1', 'industry': '0', 'registration_type': '0', 'taxpayer_nature': '0', 'upload_year': 2022, 'uploaded_files': '["企业所得税年度纳税申报表_2021.xlsx", "企业所得税年度纳税申报表_2022.xlsx", "增值税纳税申报表_2021.xls", "增值税纳税申报表_2022.xls", "财务报表_2021.xlsx", "财务报表_2022.xlsx"]', 'created_at': datetime.datetime(2024, 12, 5, 18, 58, 34)}

In [3]:
cursor = connection.cursor()

# Get column metadata
cursor.execute(f"DESCRIBE {table_name}")
columns = cursor.fetchall()

# Display column details
print("Column Details:")
for column in columns:
    print(f"Name: {column[0]}, Type: {column[1]}")

cursor.close()


Column Details:
Name: id, Type: int
Name: company_name, Type: varchar(225)
Name: tax_number, Type: varchar(225)
Name: post_data, Type: varchar(225)
Name: post_initiator_user_id, Type: int
Name: status, Type: tinyint(1)
Name: query_result, Type: varchar(225)
Name: index_standard_type, Type: varchar(225)
Name: industry, Type: varchar(225)
Name: registration_type, Type: varchar(225)
Name: taxpayer_nature, Type: varchar(225)
Name: upload_year, Type: int
Name: uploaded_files, Type: json
Name: created_at, Type: datetime


True

In [11]:
import random
from faker import Faker
import json
from datetime import datetime, timedelta

# Initialize Faker for random data generation
faker = Faker()

# Predefined uploaded files for consistency
uploaded_files = [
    "企业所得税年度纳税申报表_2021.xlsx", 
    "企业所得税年度纳税申报表_2022.xlsx", 
    "增值税纳税申报表_2021.xls", 
    "增值税纳税申报表_2022.xls", 
    "财务报表_2021.xlsx", 
    "财务报表_2022.xlsx"
]

# Function to generate random data
def generate_record(record_id):
    # Random values for fields
    company_name = faker.company()  # Random company name or empty
    tax_number = f"TaxpayerId{random.randint(1000, 9999)}"
    post_data = {
        "dateSource": random.choice([0, 1]),  # Random dateSource
        "dateType": random.choice([0, 1]),    # Random dateType
        "year": random.choice([2022, 2021, 2020]),  # Random year
        "taxpayerNo": tax_number,
        "timestamp": (datetime.now() - timedelta(seconds=random.randint(1000, 1000000))).isoformat()  # Random timestamp
    }
    post_data_json = json.dumps(post_data)
    post_initiator_user_id = random.randint(4, 5)  # Random user ID between 1 and 10
    status = random.choice([0, 1])  # Random status (0 or 1)
    query_result = None  # Keeping it as None, as in the example
    index_standard_type = random.choice(["0", "1"])  # Random choice of standard type
    industry = random.choice([ "0", "1", "2"])  # Random industry code
    registration_type = random.choice(["0", "1"])  # Random registration type
    taxpayer_nature = random.choice(["0", "1"])  # Random taxpayer nature
    upload_year = random.choice([2022, 2023, 2024])  # Random upload year

    # Current timestamp for created_at
    created_at = datetime.now()

    # Construct the record
    record = {
        "id": record_id,
        "company_name": company_name,
        "tax_number": tax_number,
        "post_data": post_data_json,
        "post_initiator_user_id": post_initiator_user_id,
        "status": status,
        "query_result": query_result,
        "index_standard_type": index_standard_type,
        "industry": industry,
        "registration_type": registration_type,
        "taxpayer_nature": taxpayer_nature,
        "upload_year": upload_year,
        "uploaded_files": json.dumps(uploaded_files),
        "created_at": created_at
    }
    return record

# Generate 10 records
generated_records = [generate_record(i + 1) for i in range(10)]

# Display generated records
for record in generated_records:
    print(record)


{'id': 1, 'company_name': 'Gonzalez-Smith', 'tax_number': 'TaxpayerId9815', 'post_data': '{"dateSource": 0, "dateType": 0, "year": 2021, "taxpayerNo": "TaxpayerId9815", "timestamp": "2024-12-04T08:51:00.080489"}', 'post_initiator_user_id': 4, 'status': 0, 'query_result': None, 'index_standard_type': '0', 'industry': '0', 'registration_type': '0', 'taxpayer_nature': '1', 'upload_year': 2022, 'uploaded_files': '["\\u4f01\\u4e1a\\u6240\\u5f97\\u7a0e\\u5e74\\u5ea6\\u7eb3\\u7a0e\\u7533\\u62a5\\u8868_2021.xlsx", "\\u4f01\\u4e1a\\u6240\\u5f97\\u7a0e\\u5e74\\u5ea6\\u7eb3\\u7a0e\\u7533\\u62a5\\u8868_2022.xlsx", "\\u589e\\u503c\\u7a0e\\u7eb3\\u7a0e\\u7533\\u62a5\\u8868_2021.xls", "\\u589e\\u503c\\u7a0e\\u7eb3\\u7a0e\\u7533\\u62a5\\u8868_2022.xls", "\\u8d22\\u52a1\\u62a5\\u8868_2021.xlsx", "\\u8d22\\u52a1\\u62a5\\u8868_2022.xlsx"]', 'created_at': datetime.datetime(2024, 12, 6, 10, 22, 33, 80522)}
{'id': 2, 'company_name': 'Stanton-Thomas', 'tax_number': 'TaxpayerId7693', 'post_data': '{"dateSourc

In [12]:
import pandas as pd

# Save to CSV
df = pd.DataFrame(generated_records)
df.to_csv("sample_data.csv", index=False)
print("Sample data saved to sample_data.csv")


Sample data saved to sample_data.csv


In [24]:
import random
from faker import Faker
from datetime import datetime, timedelta
import json

faker = Faker()

# Predefined tax numbers and company names for generated data
tax_numbers = [f"TaxpayerId{100 + i}" for i in range(10)]
company_names = [faker.company() for _ in range(10)]

# Generate 10 `CompanyInfo` records
company_info_list = [
    {
        "id": i + 100,
        "company_name": company_names[i],
        "tax_number": tax_numbers[i],
        "index_standard_type": random.choice(["1", "2", ""]),
        "industry": random.choice(["0", "1", "2"]),
        "registration_type": random.choice(["0", "1", "2"]),
        "taxpayer_nature": random.choice(["0", "1", "2"]),
        "upload_year": random.choice([2022, 2023]),
        "uploaded_files": json.dumps([
            "企业所得税年度纳税申报表_2021.xlsx",
            "企业所得税年度纳税申报表_2022.xlsx",
            "增值税纳税申报表_2021.xls",
            "增值税纳税申报表_2022.xls",
            "财务报表_2021.xlsx",
            "财务报表_2022.xlsx"
        ]),
        "post_data": json.dumps({
            "dateSource": random.choice([0, 1]),
            "dateType": random.choice([0, 1]),
            "year": 2022,
            "taxpayerNo": tax_numbers[i],
            "timestamp": datetime.now().isoformat(),
        }),
        "post_initiator_user_id": random.randint(1, 5),
        "status": random.choice([True, False]),
        "query_result": None,
        "created_at": datetime.now()
    }
    for i in range(10)
]

# Generate 10 `CompanyReport` records
company_reports = [
    {
        "id": i + 100,
        "user_id": random.randint(1, 5),  # Random user ID
        "company_tax_number": tax_numbers[i],
        "report_type": "annual",
        "year": 2022,
        "month": None,
        "quarter": None,
        "report_data": json.dumps({
            "riskMain": {
                "id": 4965 + i,
                "tenant": f"{1000 + random.randint(1, 20)}",
                "dateTime": 2022,
                "noRiskNum": random.randint(30, 40),
                "lowRiskNum": random.randint(1, 5),
                "midRiskNum": random.randint(5, 10),
                "totalScore": f"{random.uniform(70.0, 80.0):.2f}",
                "highRiskNum": random.randint(5, 15),
                "queryCondition": 20220100,
                "abnormalRiskNum": random.randint(5, 10)
            }
        }),
        "created_at": datetime.now() - timedelta(days=random.randint(0, 365)),
        "updated_at": None
    }
    for i in range(10)
]

# Print generated data
print("Generated Company Info Records:")
for info in company_info_list:
    print(info)

print("\nGenerated Company Report Records:")
for report in company_reports:
    print(report)


Generated Company Info Records:
{'id': 100, 'company_name': 'Davis-Walker', 'tax_number': 'TaxpayerId100', 'index_standard_type': '', 'industry': '2', 'registration_type': '0', 'taxpayer_nature': '0', 'upload_year': 2023, 'uploaded_files': '["\\u4f01\\u4e1a\\u6240\\u5f97\\u7a0e\\u5e74\\u5ea6\\u7eb3\\u7a0e\\u7533\\u62a5\\u8868_2021.xlsx", "\\u4f01\\u4e1a\\u6240\\u5f97\\u7a0e\\u5e74\\u5ea6\\u7eb3\\u7a0e\\u7533\\u62a5\\u8868_2022.xlsx", "\\u589e\\u503c\\u7a0e\\u7eb3\\u7a0e\\u7533\\u62a5\\u8868_2021.xls", "\\u589e\\u503c\\u7a0e\\u7eb3\\u7a0e\\u7533\\u62a5\\u8868_2022.xls", "\\u8d22\\u52a1\\u62a5\\u8868_2021.xlsx", "\\u8d22\\u52a1\\u62a5\\u8868_2022.xlsx"]', 'post_data': '{"dateSource": 1, "dateType": 0, "year": 2022, "taxpayerNo": "TaxpayerId100", "timestamp": "2024-12-06T13:51:19.651331"}', 'post_initiator_user_id': 5, 'status': True, 'query_result': None, 'created_at': datetime.datetime(2024, 12, 6, 13, 51, 19, 651348)}
{'id': 101, 'company_name': 'Myers, Wagner and Griffin', 'tax_number

In [18]:
import sys
import os

# Add the directory containing model.py to the system path
model_dir = "../"
sys.path.append(model_dir)

# Now you can import the models
from models import Base, CompanyInfo, CompanyReport

In [19]:
from sqlalchemy import create_engine
from sqlalchemy.orm import sessionmaker
from models import Base, CompanyInfo, CompanyReport  # Import your models

# Database URL
DATABASE_URL = "mysql+mysqlconnector://root:password@localhost:3306/yianshui"

# Create engine and session
engine = create_engine(DATABASE_URL)
SessionLocal = sessionmaker(autocommit=False, autoflush=False, bind=engine)
session = SessionLocal()

# Create tables if they don't exist
Base.metadata.create_all(bind=engine)


In [25]:
from datetime import datetime

# Example data from the generated lists
def insert_sample_data():
    try:
        # Insert CompanyInfo records
        for company in company_info_list:
            new_company = CompanyInfo(
                id=company["id"],
                company_name=company["company_name"],
                tax_number=company["tax_number"],
                index_standard_type=company["index_standard_type"],
                industry=company["industry"],
                registration_type=company["registration_type"],
                taxpayer_nature=company["taxpayer_nature"],
                upload_year=company["upload_year"],
                uploaded_files=company["uploaded_files"],
                post_data=company["post_data"],
                post_initiator_user_id=company["post_initiator_user_id"],
                status=company["status"],
                query_result=company["query_result"],
                created_at=company["created_at"]
            )
            session.add(new_company)

        # Insert CompanyReport records
        for report in company_reports:
            new_report = CompanyReport(
                id=report["id"],
                user_id=report["user_id"],
                company_tax_number=report["company_tax_number"],
                report_type=report["report_type"],
                year=report["year"],
                month=report["month"],
                quarter=report["quarter"],
                report_data=report["report_data"],
                created_at=report["created_at"],
                updated_at=report["updated_at"]
            )
            session.add(new_report)

        # Commit all changes
        session.commit()
        print("Sample data inserted successfully.")

    except Exception as e:
        session.rollback()
        print(f"Error inserting data: {e}")

    finally:
        session.close()

# Insert data
insert_sample_data()


Sample data inserted successfully.


In [ ]:
import os
import sys

# Dynamically find the directory of model.py
current_dir = os.getcwd()  # Get current working directory
model_file = os.path.join(current_dir, "model.py")

if os.path.exists(model_file):
    sys.path.append(current_dir)
    from models import Base, CompanyInfo, CompanyReport
else:
    print("model.py not found. Check the file path.")
